src/Microsoft.AspNet.Mvc.Core/ModelBinders/BodyModelBinder.cs

```diff
@@ -51,8 +51,13 @@ public BodyModelBinder()
             {
                 var model = await formatter.ReadAsync(formatterContext);
 
-                // key is empty to ensure that the model name is not used as a prefix for validation.
-                return new ModelBindingResult(model, key: string.Empty, isModelSet: true);
+                var isTopLevelObject = bindingContext.ModelMetadata.ContainerType == null;
+
+                // For compatibility with MVC 5.0 for top level object we want to consider an empty key instead of 
+                // the parameter name/a custom name. In all other cases (like when binding body to a property) we
+                // consider the entire ModelName as a prefix.
+                var modelBindingKey = isTopLevelObject ? string.Empty : bindingContext.ModelName;
```

Comment:
This seems like to me like the opposite of what we discussed in the meeting.  (Using the empty key for a top-level-model). Is this truly a compat scenario with MVC5?